client/src/pages/AddFood.js

```diff
@@ -21,59 +20,76 @@ function AddFood() {
     })
     .required();
   const { register, handleSubmit, reset } = useForm({
-    resolver: yupResolver(schema),
+    resolver: null,
   });
 
   const url = "http://localhost:5000/food";
 
   const mutation = useMutation({
-    mutationFn: (newFood) => {
-      return axios.post(url, newFood).then((response) => {
-        setFood(response.data);
-      });
+    mutationFn: async (newFood) => {
+      console.log(newFood);
+      const config = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newFood),
+      };
+      const response = await fetch(url, config);
+      if (response.ok) {
+        console.log(response);
+        return await response.json();
+      }
     },
   });
 
-  function handleForm(e) {
-    e.preventDefault();
-    mutation.mutate(url, { title: "title", img: "img", author: "author" });
+  function handleForm(data) {
+    console.log(data);
+    mutation.mutate({
+      title: data.title,
+      img: data.img,
+      author: data.author,
+    });
   }
 
-  function handleChange(e) {
-    setFood(e.target.value);
-    console.log(e);
-  }
+  useEffect(() => {
+    reset();
```

Comment:
This seems reasonable.  Was react-hook-form holding onto the state from the previous food item?

Reply:
Yes ,but I'm currently only resetting the last entry